flatcar-linux.json

```diff
@@ -0,0 +1,63 @@
+{
+    "variables": {
+        "flatcar_channel": "alpha",
+        "flatcar_version": "current",
+        "iso_checksum": "",
+        "iso_checksum_type": "none",
+        "disk_size": "40000",
+        "memory": "2048M",
+        "boot_wait": "45s",
+        "headless": "false"
+    },
+
+    "builders": [
+        {
+            "type": "qemu",
+            "iso_url": "https://{{user `flatcar_channel` }}.release.flatcar-linux.net/amd64-usr/current/flatcar_production_iso_image.iso",
+            "iso_checksum": "{{ user `iso_checksum` }}",
+            "iso_checksum_type": "{{ user `iso_checksum_type` }}",
+            "output_directory": "builds",
+            "shutdown_command": "sudo shutdown now",
+            "disk_size": "{{ user `disk_size` }}",
+            "format": "qcow2",
+            "accelerator": "kvm",
+            "vm_name": "flatcar-linux-{{user `flatcar_channel` }}.qcow2",
+            "boot_wait": "{{user `boot_wait` }}",
+            "ssh_username": "core",
+            "ssh_password": "packer",
+            "headless": "{{ user `headless` }}",
+            "boot_command": [
+                "sudo passwd core<enter><wait>",
+                "packer<enter>",
+                "packer<enter>",
+                "sudo systemctl start sshd.service<enter>"
+            ],
+            "qemuargs": [
+                [ "-m", "{{user `memory` }}" ]
+            ]
+        }
+    ],
+
+    "provisioners": [
+        {
+            "type": "file",
+            "source": "ignition.json",
+            "destination": "/tmp/ignition.json"
+        },
+        {
+            "type": "shell",
+            "inline": [
+                "sudo flatcar-install -d /dev/vda -C {{ user `flatcar_channel` }} -i /tmp/ignition.json"
+            ]
+        }
+    ],
+
+    "post-processors": [
+        {
+            "type": "vagrant",
```

Comment:
Can we make the post-processor to generate and include a metadata.json file to avoid

```
$ vagrant box add builds/flatcar-alpha-current-libvirt.box
==> box: Box file was not detected as metadata. Adding it directly...
A name is required when adding a box file directly. Please pass
the `--name` parameter to `vagrant box add`. See
`vagrant box add -h` for more help.
```

Reply:
`vagrant box add builds/flatcar-alpha-current-libvirt.box` fails but should work, I think.

Reply:
I'm not sure that feature exists though, generating a metadata.json file when using vagrant as type. Someone asked for similar feature here https://github.com/hashicorp/packer/issues/5833. If you add the box directly from Vagrant cloud, there won't be issues with it not detecting the box file. Except we want to self host our boxes eventually, manually making a metadata.json file might not be necessary.

Reply:
OK, had not seen that issue, fair enough.Let's go without :)